openghg/util/_data_level.py

```diff
@@ -43,16 +43,17 @@ def format_data_level(data_level: Optional[str]) -> Optional[str]:
     msg_expected = "Expect: '0', '1', '2', '3' (or decimal to indicate sub-level)"
 
     if data_level is not None:
-        data_level = str(data_level)  # in case a int/float is passed - cast to str
         if not is_number(data_level):
             msg = f"Unable to interpret data_level input: {data_level}. " + msg_expected
             logger.error(msg)
             raise MetadataFormatError(msg)
 
-        number = float(data_level)
-        if number >= 4:
-            msg = "data_level input contains a number > 3: {data_level}. " + msg_expected
+        data_level_check = float(data_level)
+        if data_level_check < 0 or data_level_check >= 4:
+            msg = f"Invalid input for data_level: {data_level}. " + msg_expected
             logger.error(msg)
```

Comment:
Do want to log and raise an exception? If we don't plan on handling this exception, then it will give the same info. (OTOH, if we recorded logs somewhere, it might be useful for debugging to log errors, since we wouldn't need people to send us tracebacks.)

Reply:
Yeah there's a few places in the code where we're just including `logger.error` rather than `logger.exception()` and raising an error. I kept this with what was written for the moment but we can switch over to the `logger.exception` instead.

Reply:
Actually I just changed it as I think we should move over to not just logging errors going forward.

Reply:
Okay sounds good. I guess recording logs could be useful, but seems like a big project, and it seems pretty standard for other python projects to just raise an issue with the traceback.